src/test/ui/consts/array-to-slice-cast.rs

```diff
@@ -0,0 +1,8 @@
+// check-pass
+
+fn main() {}
+
+const fn foo() {
+    let x = [1, 2, 3, 4, 5];
+    let y: &[_] = &x;
```

Comment:
Could you add a test with something like `(bool, [i32])`.

Reply:
No

```
error[E0658]: unsized tuple coercion is not stable enough for use and is subject to change
 --> src/main.rs:3:27
  |
3 |     let y: &(u32, [u8]) = &x;
  |                           ^^
  |
  = note: see issue #42877 <https://github.com/rust-lang/rust/issues/42877> for more information
```

but I'll figure out an example with structs